
TENSORFLOW USERS:
You must use the python3.6-anaconda/5.2.0 module to run TensorFlow.
See https://arc-ts.umich.edu/greatlakes/software/tensorflow for more
information.



The following have been reloaded with a version change:
  1) cuda/10.0.130 => cuda/10.1.105


Currently Loaded Modules:
  1) python3.7-anaconda/2019.07   2) cudnn/10.0-v7.6   3) cuda/10.1.105

 

mkdir: cannot create directory ‘results/9296214’: File exists
mkdir: cannot create directory ‘results/9296214/log’: File exists
2020-07-15 23:36:47.227586: E tensorflow/stream_executor/cuda/cuda_driver.cc:313] failed call to cuInit: UNKNOWN ERROR (303)
Test  329
2020-07-15 23:36:46.746975

How often each label is chosen for non-IID data
[20, 18, 26, 18, 14, 19, 19, 20, 21, 25]

Schema 1: Each client partially IID
--------------------------------------------------
percent data distributed IID:  80
number of classes for non-IID data:  2
data points per client (mean, std dev): ( 600 ,  0 )
--------------------------------------------------
round  1, metrics=<sparse_categorical_accuracy=0.18424999713897705,loss=2.2766685485839844>
13.3767 seconds
Tested. Sparse categorical accuracy: 33.33

round  2, metrics=<sparse_categorical_accuracy=0.3503333330154419,loss=2.1341452598571777>
9.1302 seconds
Tested. Sparse categorical accuracy: 60.00

round  3, metrics=<sparse_categorical_accuracy=0.5227500200271606,loss=1.6145461797714233>
9.1619 seconds
Tested. Sparse categorical accuracy: 66.67

round  4, metrics=<sparse_categorical_accuracy=0.7293333411216736,loss=0.922972559928894>
10.2139 seconds
Tested. Sparse categorical accuracy: 80.00

round  5, metrics=<sparse_categorical_accuracy=0.8365833163261414,loss=0.5263684988021851>
9.0458 seconds
Tested. Sparse categorical accuracy: 100.00

round  6, metrics=<sparse_categorical_accuracy=0.8846666812896729,loss=0.3939875066280365>
9.8170 seconds
Tested. Sparse categorical accuracy: 93.33

Final confusion matrix
Traceback (most recent call last):
  File "R_tff_main.py", line 28, in <module>
    p1.go(test, batch)
  File "/home/apawlik/federated/R_1_each_client_partially_iid.py", line 208, in go
    self.train(num, batch, 1)
  File "/home/apawlik/federated/R_partitioner.py", line 320, in train
    print(confusion_matrix(actuals, preds, labels = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))
  File "/sw/arcts/centos7/python3.7-anaconda/2019.07/lib/python3.7/site-packages/sklearn/metrics/classification.py", line 253, in confusion_matrix
    y_type, y_true, y_pred = _check_targets(y_true, y_pred)
  File "/sw/arcts/centos7/python3.7-anaconda/2019.07/lib/python3.7/site-packages/sklearn/metrics/classification.py", line 71, in _check_targets
    check_consistent_length(y_true, y_pred)
  File "/sw/arcts/centos7/python3.7-anaconda/2019.07/lib/python3.7/site-packages/sklearn/utils/validation.py", line 205, in check_consistent_length
    " samples: %r" % [int(l) for l in lengths])
ValueError: Found input variables with inconsistent numbers of samples: [10000, 15]
